### PR TITLE
Break up theme.liquid into separate snippets

### PR DESCRIPTION
**Pull request by graygilmore**
This file was getting a little verbose and hard to follow. By separating
each of these components into their own snippets we can easily see where
things begin and end.

Note: no code was actually modified here, this is purely an organizational PR.

### CODE REVIEW

**Comment by Cam** [comment]
Good call. I do this right at the beginning, every time I start a new Timber based project. Snippets FTW.

**Comment by graygilmore** [comment]
@cshold any thoughts? This is the first thing I do, too. It'd be great if we could have folder structure inside of `/snippets`, but of course that is well outside the scope of this project.

**Comment by cshold** [comment]
For developers snippets are a huge win. Organization, cleanliness, both great. Not too long ago we in-lined snippets that were only used once (like the header and footer) so merchants would have an easier time finding places to edit or paste code from plugins/apps/docs. Editing markup is already tricky enough, and the added reliance on snippets created more of a hurdle for them.

Unfortunately we don't have stats on whether or not this is helping and we need to do some UX testing to see if it makes a noticeable difference. For the time being, we're only going to use snippets for code that is used multiple times, though it isn't set in stone forever.

**Comment by Cam** [comment]
That makes total sense from the end client point of view. It's only takes 5 minutes to move various bits into snippets.

**Comment by graygilmore** [comment]
I'd definitely be interested in knowing if customers are using Timber themselves or getting a developer to set it up for them. If a developer sets it up, I'm guessing they're just going to separate into snippets anyways.

Either way, a good enough reason to close the PR. Would love to hear about any stats you guys learn about, though!
